### PR TITLE
add UniV3Adapter02 with min arb params in uniV3Swap function

### PR DESCRIPTION
**Pull request by mikec**

### CODE REVIEW

File: contracts/adapter/UniV3Adapter02.sol [hunk 1]
```diff
@@ -0,0 +1,105 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity =0.8.10;
+pragma abicoder v1;
+
+import '../token/IERC20.sol';
+import '../token/IWETH.sol';
+
+import 'hardhat/console.sol';
```
Comment (Kyrrui) [inline contracts/adapter/UniV3Adapter02.sol]: This should be removed

File: contracts/adapter/UniV3Adapter02.sol [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    weth = _weth;
+  }
+
+  /// @dev Makes a call to the Uniswap V3SwapRouter with swap byte data
+  /// @dev returns the requested tokenOutAmount to Account and keeps the rest
+  /// @param data swap byte data for Uniswap V3SwapRouter
+  /// @param tokenIn Address of the token to be swapped
+  /// @param tokenOut Address of the token to be returned from the swap
+  /// @param tokenOutAmount Amount of tokenOut to transfer to account
+  /// @param account Address of the account to receive the tokenOut
```
Comment (Kyrrui) [inline contracts/adapter/UniV3Adapter02.sol:45]: Add mintokeninarb and mintokenoutarb param comments

File: contracts/adapter/UniV3Adapter02.sol [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+      uint ethBalRemaining = address(this).balance;
+      if (ethBalRemaining < minTokenOutArb) {
+        revert NotEnoughTokenOut(ethBalRemaining);
+      }
+      ADAPTER_OWNER.transfer(ethBalRemaining);
+    } else {
+      tokenOut.transfer(account, tokenOutAmount);
+      uint tokenOutBalRemaining = tokenOut.balanceOf(address(this));
+      if (tokenOutBalRemaining < minTokenOutArb) {
+        revert NotEnoughTokenOut(tokenOutBalRemaining);
```
Comment (Kyrrui) [inline contracts/adapter/UniV3Adapter02.sol:79]: Might consider having a different error for not reaching arb

File: contracts/adapter/UniV3Adapter02.sol (hunk 1 above)
Comment (mikec) [reply contracts/adapter/UniV3Adapter02.sol]: oops! thanks

**Pull request merged**